brave/brave-core
# Brave Ads server requests should not retry if a browser upgrade is required

Resolves https://github.com/brave/brave-browser/issues/23021

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Confirm HTTP_UPGRADE_REQUIRED does not retry for issuers
- Confirm HTTP_UPGRADE_REQUIRED does not retry for catalog
- Confirm HTTP_UPGRADE_REQUIRED does not retry for redeeming unblinded payment token
- Confirm HTTP_UPGRADE_REQUIRED does not retry for redeeming unblinded payment tokens
- Confirm HTTP_UPGRADE_REQUIRED does not retry for subdivision targeting
- Confirm HTTP_BAD_REQUEST does not retry for redeeming unblinded token
- Confirm Issuers retry for non HTTP_OK
- Confirm catalog retry for non HTTP_OK
- Confirm redeeming unblinded payment token retry for HTTP_NOT_FOUND
- Confirm redeeming unblinded payment token retry for HTTP_ACCEPTED
- Confirm redeeming unblinded payment token retry for non HTTP_OK
- Confirm redeeming unblinded payment tokens retry for non HTTP_OK
- Confirm subdivision targeting retry for non HTTP_OK
